pkg/status/status-controller.go

```diff
@@ -333,10 +333,13 @@ func (c *Controller) runWorkStatusInformer(ctx context.Context) {
 	// add the event handler functions
 	c.workStatusInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
+			if shouldSkipAdd(c.wdsName, obj) {
```

Comment:
Why not a shared ShouldSkip() function that all call and just decide if to skip or not (without performing the action)

Reply:
I'm trying to follow here: What action is performed here?

Reply:
@waltforme Regarding the action - you are correct. 
Regarding the sharing?  ShouldAdd and ShouldDelete seems identical.  And also the first section of shouldUpdate. 
Maybe just add IsInThisWDS()   and have the delete and Add use this directly, and the update can use 
`If !IsInThisWDS() || SkipUpdate() { return}`
Not a big issue - if you decide you don't agree - I'm fine.  I just don't like identical implementation in different functions.

Reply:
I got the point. Thanks! And revised as suggested.